runtime/vm/jnicsup.cpp

```diff
@@ -1660,7 +1660,7 @@ j9jni_deleteGlobalRef(JNIEnv *env, jobject globalRef, jboolean isWeak)
 #endif
 
 #if defined(J9VM_GC_REALTIME)
-		if (J9_EXTENDED_RUNTIME_USER_REALTIME_ACCESS_BARRIER == (vm->extendedRuntimeFlags & J9_EXTENDED_RUNTIME_USER_REALTIME_ACCESS_BARRIER)) {
+		if (J9_ARE_ALL_BITS_SET(vm->extendedRuntimeFlags, J9_EXTENDED_RUNTIME_USER_SATB_ACCESS_BARRIER)) {
```

Comment:
This seems to be the only spot that makes use of this flag. I don't really see what was different about this check over checking a barrier type (now, with 2 SATB barrier types, this check is marginally faster, but that's negligible argument). 

There is a potential we could convert this check, and free up one extended runtime flag (which I believe is scarce resource)

Reply:
This is the only spot that makes use of this runtime flag. I agree, we can remove it  and rely on the barrier type. 

I've made the following changes for this:

`j9consts.h`
_Changed flag define to:_
```C++
#define J9_EXTENDED_RUNTIME_UNUSED_0x80 0x80
```
`jnicsup.cpp`
_Check Barrier type rather than Runtime Flag:_
```C++
if ((J9_GC_WRITE_BARRIER_TYPE_SATB == vm->gcWriteBarrierType) || (J9_GC_WRITE_BARRIER_TYPE_SATB_AND_OLDCHECK == vm->gcWriteBarrierType)) {
```
`MetronomeDelegate.cpp`
_Remove this line for setting the runtime flag:_
```C++
javaVM->extendedRuntimeFlags |= J9_EXTENDED_RUNTIME_USER_SATB_ACCESS_BARRIER;
```

Reply:
Actually, we can go a step further and remove the check all together, we can call the api unconditionally (base class is no-op and only implementer is Metronome Access Barrier).

Is this okay?

Reply:
Yeah, seems like we can remove this check here and rely on virtual dispatch. But, we also miss implementation for Standard SATB variant (so, we need an _if_ check in StandarardAB::jniDeleteGlobalReference).

Reply:
check has been removed